File: component/src/main/java/com/mackwu/component/ui/view/MyView.java
package com.mackwu.component.ui.view;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatImageView;

import com.mackwu.base.util.Logger;
import com.mackwu.component.R;


/**
 * ===================================================
 * Created by MackWu on 2022/2/22 17:47
 * <a href="mailto:wumengjiao828@163.com">Contact me</a>
 * <a href="https://github.com/mackwu828">Follow me</a>
 * ===================================================
 */
public class MyView extends AppCompatImageView {

    Paint paint;

    public MyView(Context context) {
        this(context, null);
    }

    public MyView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public MyView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        initView();
    }

    private void initView() {
        paint = new Paint();
        paint.setAntiAlias(true);
        paint.setColor(Color.BLACK);
        paint.setStrokeWidth(10f);
        paint.setStyle(Paint.Style.STROKE);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
    }

}

File: kt/src/main/java/com/mackwu/kt/pattern/observer/onclick/MyView.kt
package com.mackwu.kt.pattern.observer.onclick

/**
 * ===================================================
 * Created by MackWu on 2019/12/23 19:34
 * <a href="mailto:wumengjiao828@163.com">Contact me</a>
 * <a href="https://github.com/mackwu828">Follow me</a>
 * ===================================================
 * view是被观察者。当点击时(状态发生变化)，去通知所有的OnClickListener，这里只有一个。
 */
class MyView{

    fun setOnClickListener(l: OnClickListener) {
        l.onClick(this)
    }
}

//fun main() {
//    val myView = MyView()
//    myView.setOnClickListener(object : OnClickListener {
//        override fun onClick(v: MyView) {
//
//        }
//    })
//}
